### PR TITLE
Implementation of dummy algorythm for stemma generation

### PR DESCRIPTION
**Pull request by SirReginald1**
Finished classes and tests for:

- Base stemma algorythmes
- Dummy algorythm
- Implemented stemma construction from stemma algorythm in stemma class.

### CODE REVIEW

File: src/stemmabench/algorithms/stemma_algorithm.py [hunk 1]
```diff
@@ -1,2 +1,110 @@
...
+    """Base class for all th stemma algorithmes.
+
+    ### Attributes:
+        - folder_path (str): The path to the folder containing all the texts.
+        - manuscripts (dict): The dictionay of all the texts with text labels as keys and texts as values.
+        This is what is ued to build the stemmas.
+    """
+
+    def __init__(self, 
+                 folder_path: Union[str, None] = None) -> None:
```
Comment (SphRbtHyk) [inline src/stemmabench/algorithms/stemma_algorithm.py:16]: Remove all need for folder_path

File: src/stemmabench/algorithms/stemma_algorithm.py [hunk 2]
```diff
@@ -1,2 +1,110 @@
...
+        """Builds a list representation of stemma tree edges.
+        This is where the algorythmes are implemented.
+
+        ### Returns:
+            - list: List of all the edges from the stemma tree constructed by the algorythm.
+
+        ### Raises:
+            - NotImplementedError: Method not implemented for this class.
+        """
+        raise NotImplementedError()
```
Comment (SphRbtHyk) [inline src/stemmabench/algorithms/stemma_algorithm.py:85]: Remove parenthesis when no error message

File: src/stemmabench/algorithms/stemma_dummy.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    ### Attributes:
+        - folder_path (str): The path to the folder containing all the texts.
+        - manuscripts (dict): The dictionay of all the texts with text labels as keys and texts as values.
+        - width (int): The number of children generated for each manuscript. May be overriten by more complex generation parameters in future implementations.
+        - seed (int): The seed used to produce reproducible pseudorandom results.
+        - generator (numpy.random._generator.Generator): The generator used to generate random numbers.
+    """
+
+    def __init__(self,
+                 folder_path: Union[str,None] = None,
```
Comment (SphRbtHyk) [inline src/stemmabench/algorithms/stemma_dummy.py:21]: Remove folder_path from arguments

File: tests/test_stemma_dummy.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    """
+    
+    def setUp(self) -> None:
+        """Setup the unit test.
+        """
+        self.stemma_folder_path = "tests/test_data/test_stemma"
+        self.stemmaDummy_empty = StemmaDummy()
+        self.stemmaDummy = StemmaDummy(self.stemma_folder_path, width=5)
+        self.stemmaDummy_seed = StemmaDummy(folder_path=self.stemma_folder_path, seed=1)
+        self.stemmaDummy.compute()
```
Comment (SphRbtHyk) [inline tests/test_stemma_dummy.py:19]: this is what you want to test ;) 
should not be instantiated in the setup !

File: tests/test_stemma_dummy.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        self.test_random_levels = [['3'], 
+                                   ['4', '7'], 
+                                   ['9', '1', '11', '6'], 
+                                   ['10', '12', '13', '2', '5', '8']]
+        self.test_edges_list = [['3', '4'],['3', '7'],['4', '9'],
+                                ['4', '1'],['7', '11'],['7', '6'],
+                                ['9', '10'],['9', '12'],['1', '13'],
+                                ['1', '2'],['11', '5'],['11', '8']]
+
+    def test_StemmaDummy(self):
```
Comment (SphRbtHyk) [inline tests/test_stemma_dummy.py:29]: majuscule in test name :)

File: src/stemmabench/algorithms/stemma_algorithm.py [hunk 6]
```diff
@@ -1,2 +1,110 @@
...
+            - folder_path (str, Optional): The path to the folder containing the texts. The path specified here will surplant the previous path defined in constructor and will be set as new path_folder attribute.
+            !!! All .txt files in this folder must be files containing Manuscript texts unless the file name contains the substring "edge" !!!
+        
+        ### Returns:
+            - Manuscript: The root of the stemma with the rest of its tree as its children.
+
+        ### Raises:
+            - ValueError: If both the folder_path parameter and the object _folder_path attribute have not been specified.
+        """
+        if folder_path == None and self.folder_path == None:
```
Comment (SphRbtHyk) [inline src/stemmabench/algorithms/stemma_algorithm.py:71]: change all of this by raise NotImplementedError

File: src/stemmabench/algorithms/stemma_algorithm.py (hunk 1 above)
Comment (SirReginald1) [reply src/stemmabench/algorithms/stemma_algorithm.py:16]: Removed

File: src/stemmabench/algorithms/stemma_dummy.py (hunk 3 above)
Comment (SirReginald1) [reply src/stemmabench/algorithms/stemma_dummy.py:21]: Done

File: src/stemmabench/algorithms/stemma_dummy.py [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+    @property
+    def width(self):
+        return self._width
+    
+    @property
+    def generator(self):
+        return self._generator
+    
+    def compute(self, folder_path: Union[str, None] = None, width: Union[int, None] = None) -> Manuscript:
```
[the remark this answers is not in the document]
Comment (SirReginald1) [reply src/stemmabench/algorithms/stemma_dummy.py:49]: I left it in for now as the object needs some way of accessing the list of manuscripts it needs to build the stemma.

File: tests/test_stemma_dummy.py (hunk 4 above)
Comment (SirReginald1) [reply tests/test_stemma_dummy.py:19]: Removed

File: tests/test_stemma_algorithm.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+class TestStemmaAlgo(unittest.TestCase):
+    """Unit tests for the StemmaAlgo class.
+    """
+    
+    def setUp(self) -> None:
+        """Setup the unit test.
+        """
+        self.stemma_folder_path = "tests/test_data/test_stemma"
+        self.stemmaAlgo_empty = StemmaAlgo()
```
[the remark this answers is not in the document]
Comment (SirReginald1) [reply tests/test_stemma_algorithm.py:16]: Removed

File: tests/test_stemma_dummy.py (hunk 5 above)
Comment (SirReginald1) [reply tests/test_stemma_dummy.py:29]: Put in snake case

File: src/stemmabench/algorithms/stemma_algorithm.py (hunk 6 above)
Comment (SirReginald1) [reply src/stemmabench/algorithms/stemma_algorithm.py:71]: Done

File: src/stemmabench/algorithms/stemma_algorithm.py (hunk 2 above)
Comment (SirReginald1) [reply src/stemmabench/algorithms/stemma_algorithm.py:85]: Removed

**Review by SphRbtHyk — APPROVED** [review]

**Pull request merged**